File: server/models/Candidate.js
const mongoose = require('mongoose');

const candidateSchema = new mongoose.Schema({
  candidate_id: { type: String, required: true, unique: true },
  job_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Job', required: true },
  resume_id: { type: String, required: true }, // Changed from ObjectId to String to accept UUID
  created_at: { type: Date, default: Date.now },
  match_score: { type: Number, default: 0 },
  full_name: String,
  email: String,
  phone: String,
  linkedin: [String],
  github: [String],
  summary: String,
  skills: [String],
  education: [{ degree: String, institute: String, year: String }],
  experience: [{ role: String, company: String, period: String, description: String }],
  projects: [{ title: String, description: String }],
  status: { type: String, enum: ['New', 'Shortlisted', 'Interviewed', 'Rejected', 'Hired'], default: 'New' }
});

module.exports = mongoose.model('Candidate', candidateSchema); 

File: backend/app/models/candidate.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime

class EducationEntry(BaseModel):
    degree: str
    institute: str
    year: str

class ExperienceEntry(BaseModel):
    role: str
    company: str
    period: str
    description: str

class ProjectEntry(BaseModel):
    title: str
    description: str

class Candidate(BaseModel):
    candidate_id: str  # Primary key (UUID/String)
    job_id: str        # Associated job ID
    full_name: str
    email: str
    phone: str
    linkedin: List[str] = []
    github: List[str] = []
    portfolio: List[str] = []
    summary: Optional[str] = None
    skills: List[str] = []
    education: List[EducationEntry] = []
    experience: List[ExperienceEntry] = []
    projects: List[ProjectEntry] = []
    links: List[str] = []
    match_score: Optional[float] = None
    resume_id: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None 